test/cmsapitest.c

```diff
@@ -78,6 +78,216 @@ static int test_encrypt_decrypt_aes_256_gcm(void)
+        0x09, 0x03, 0x31, 0x0b, 0x06, 0x09, 0x2a, 0x86,
+        0x48, 0x86, 0xf7, 0x0d, 0x01, 0x07, 0x01, 0x30,
+        0x1c, 0x06, 0x09, 0x2a, 0x86, 0x48, 0x86, 0xf7,
+        0x0d, 0x01, 0x09, 0x05, 0x31, 0x0f, 0x17, 0x0d,
+        0x32, 0x30, 0x31, 0x32, 0x31, 0x31, 0x30, 0x39,
+        0x30, 0x30, 0x31, 0x33, 0x5a, 0x30, 0x2f, 0x06,
+        0x09, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d, 0x01,
+        0x09, 0x04, 0x31, 0x22, 0x04, 0x20, 0xb0, 0x80,
+        0x22, 0xd3, 0x15, 0xcf, 0x1e, 0xb1, 0x2d, 0x26,
+        0x65, 0xbd, 0xed, 0x0e, 0x6a, 0xf4, 0x06, 0x53,
+        0xc0, 0xa0, 0xbe, 0x97, 0x52, 0x32, 0xfb, 0x49,
+        0xbc, 0xbd, 0x02, 0x1c, 0xfc, 0x36, 0x30, 0x0d,
+        0x06, 0x09, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d,
+        0x01, 0x01, 0x01, 0x05, 0x00, 0x04, 0x82, 0x01,
+        0x00, 0x37, 0x44, 0x39, 0x08, 0xb2, 0x19, 0x52,
+        0x35, 0x9c, 0xd0, 0x67, 0x87, 0xae, 0xb8, 0x1c,
+        0x80, 0xf4, 0x03, 0x29, 0x2e, 0xe3, 0x76, 0x4a,
+        0xb0, 0x98, 0x10, 0x00, 0x9a, 0x30, 0xdb, 0x05,
+        0x28, 0x53, 0x34, 0x31, 0x14, 0xbd, 0x87, 0xb9,
+        0x4d, 0x45, 0x07, 0x97, 0xa3, 0x57, 0x0b, 0x7e,
+        0xd1, 0x67, 0xfb, 0x4e, 0x0f, 0x5b, 0x90, 0xb2,
+        0x6f, 0xe6, 0xce, 0x49, 0xdd, 0x72, 0x46, 0x71,
+        0x26, 0xa1, 0x1b, 0x98, 0x23, 0x7d, 0x69, 0x73,
+        0x84, 0xdc, 0xf9, 0xd2, 0x1c, 0x6d, 0xf6, 0xf5,
+        0x17, 0x49, 0x6e, 0x9d, 0x4d, 0xf1, 0xe2, 0x43,
+        0x29, 0x53, 0x55, 0xa5, 0x22, 0x1e, 0x89, 0x2c,
+        0xaf, 0xf2, 0x43, 0x47, 0xd5, 0xfa, 0xad, 0xe7,
+        0x89, 0x60, 0xbf, 0x96, 0x35, 0x6f, 0xc2, 0x99,
+        0xb7, 0x55, 0xc5, 0xe3, 0x04, 0x25, 0x1b, 0xf6,
+        0x7e, 0xf2, 0x2b, 0x14, 0xa9, 0x57, 0x96, 0xbe,
+        0xbd, 0x6e, 0x95, 0x44, 0x94, 0xbd, 0xaf, 0x9a,
+        0x6d, 0x77, 0x55, 0x5e, 0x6c, 0xf6, 0x32, 0x37,
+        0xec, 0xef, 0xe5, 0x81, 0xb0, 0xe3, 0x35, 0xc7,
+        0x86, 0xea, 0x47, 0x59, 0x38, 0xb6, 0x16, 0xfb,
+        0x1d, 0x10, 0x55, 0x48, 0xb1, 0x44, 0x33, 0xde,
+        0xf6, 0x29, 0xbe, 0xbf, 0xbc, 0x71, 0x3e, 0x49,
+        0xba, 0xe7, 0x9f, 0x4d, 0x6c, 0xfb, 0xec, 0xd2,
+        0xe0, 0x12, 0xa9, 0x7c, 0xc9, 0x9a, 0x7b, 0x85,
+        0x83, 0xb8, 0xca, 0xdd, 0xf6, 0xb7, 0x15, 0x75,
+        0x7b, 0x4a, 0x69, 0xcf, 0x0a, 0xc7, 0x80, 0x01,
+        0xe7, 0x94, 0x16, 0x7f, 0x8d, 0x3c, 0xfa, 0x1f,
+        0x05, 0x71, 0x76, 0x15, 0xb0, 0xf6, 0x61, 0x30,
+        0x58, 0x16, 0xbe, 0x1b, 0xd1, 0x93, 0xc4, 0x1a,
+        0x91, 0x0c, 0x48, 0xe2, 0x1c, 0x8e, 0xa5, 0xc5,
+        0xa7, 0x81, 0x44, 0x48, 0x3b, 0x10, 0xc2, 0x74,
+        0x07, 0xdf, 0xa8, 0xae, 0x57, 0xee, 0x7f, 0xe3,
+        0x6a
+    };
+
+    ret = TEST_ptr(bio = BIO_new_mem_buf(cms_data, sizeof(cms_data)))
+          && TEST_ptr(cms = d2i_CMS_bio(bio, NULL))
+          && TEST_true(CMS_verify(cms, NULL, NULL, NULL, NULL,
+                                  CMS_NO_SIGNER_CERT_VERIFY));
+    CMS_ContentInfo_free(cms);
+    BIO_free(bio);
```

Comment:
It may make sense to repeat the test with the second `d2i_CMS_bio()` idiom. Part of the regression was that these behaved differently.

```C
    cms = NULL;
    ret = TEST_ptr(bio = BIO_new_mem_buf(cms_data, sizeof(cms_data)))
          && TEST_ptr(d2i_CMS_bio(bio, &cms))
          && TEST_true(CMS_verify(cms, NULL, NULL, NULL, NULL,
                                  CMS_NO_SIGNER_CERT_VERIFY));
    CMS_ContentInfo_free(cms);
    BIO_free(bio);
```

Reply:
The other case is what the cms app uses.. and since the cms tests hit this quite heavily then it is not needed.